src/opentimelineio/CORE_VERSION_MAP.cpp

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Contributors to the OpenTimelineIO project
+//
+// This document is automatically generated by running the `make version-map`
+// make target.  It is part of the unit tests suite and should be updated
+// whenever schema versions change.  If it needs to be updated, run: `make
+// version-map-update` and this file should be regenerated.
+//
+// This maps a "Label" to a map of Schema name to Schema version. The intent is
+// that these sets of schemas can be used for compatability with future
+// versions of OTIO, so that a newer version of OTIO can target a compatability
+// version of an older library.
+#include "opentimelineio/typeRegistry.h"
+
+namespace opentimelineio { namespace OPENTIMELINEIO_VERSION {
+
+const label_to_schema_version_map CORE_VERSION_MAP {
+    { "0.15.0.dev1",
```

Comment:
if we did that, what would we do with the dev1?

Reply:
This would need to be defined, but one possibility is to map strings to numbers (we would need a predefined list)  "dev" = 0, "alpha" = 1, "beta" = 2, etc.

Reply:
The dev1 doesn't show up anywhere outside of main branch development builds, that will disappear as part of the release process.  the released version should just be `0.15.0`.

I think its more direct to not have a key for the label if we haven't built a map for it.  We couldn't guarantee downgrading to that label if we don't have a spec or downgrade functions for it.  I'd vote against it.

Reply:
If the string can be left off, I would definitely vote for that as well as it would simplify things.

For the version string to `uint64_t` conversion there are a lot of different techniques people use but one I think tends to work well is bit shifting.  Split around the '.' and shift each number.  Something like this for example:
```
uint64_t(a, b, c) = a << 32 | b << 16 | c
```
That would allow lexicographical comparison with a max version of ("65536"."65536"."65536"), and it can be converted back into a string as well by doing the reverse operation. 

There are other ways of doing it as well, so we could have a discussion about it if we want to take this approach.

Reply:
Lets do this as a follow up conversation around how to structure labels in the future.